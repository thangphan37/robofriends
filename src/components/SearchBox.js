import React from 'react';

function SearchBox({onSearchRobots}) {
  return(
    <div className="pa2">
      <label htmlFor="robo" style={{color: '#FFFFFF', padding: '2px'}}>
        Let's type:
      </label>
      <input id="robo" type="search" className="pa3 ba b--green bg-lightest-blue" onChange={onSearchRobots} placeholder="Search robot"/>
    </div>
  )
}

export default SearchBox;