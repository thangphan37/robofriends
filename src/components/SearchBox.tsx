import React from 'react';

type Props = {
    onSearchRobots(event: React.FormEvent<HTMLInputElement>): void;
};

const SearchBox: React.FC<Props> = ({onSearchRobots}) => {
  return(
    <div className="pa2">
      <label htmlFor="robo" style={{color: '#FFFFFF', padding: '2px'}}>
        Let's type:
      </label>
      <input id="robo" type="search" className="pa3 ba b--green bg-lightest-blue" onChange={onSearchRobots} placeholder="Search robot"/>
    </div>
  )
}

export default SearchBox;