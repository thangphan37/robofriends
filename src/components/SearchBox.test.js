import React from 'react';
import renderer from 'react-test-renderer';
import SearchBox from './SearchBox';

it('snapshot SearchBox component', () => {
  const wrapper = renderer.create(<SearchBox />).toJSON();

  expect(wrapper).toMatchSnapshot();
})