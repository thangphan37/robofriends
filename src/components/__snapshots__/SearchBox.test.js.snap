// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot SearchBox component 1`] = `
<div
  className="pa2"
>
  <label
    htmlFor="robo"
    style={
      Object {
        "color": "#FFFFFF",
        "padding": "2px",
      }
    }
  >
    Let's type:
  </label>
  <input
    className="pa3 ba b--green bg-lightest-blue"
    id="robo"
    placeholder="Search robot"
    type="search"
  />
</div>
`;
